### PR TITLE
Android/source context mentions

### PR DESCRIPTION
**Pull request by stefanosiano**
## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

added source context mention to recent Android highlights, proguard and common source context

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

### CODE REVIEW

**Comment by marandaneto** [comment]
@stefanosiano please follow the guideline we agreed on, the source code context should have its own sections in the docs, similar to Flutter/RN,  the guidelines have examples of PRs and which files should be modified, not limited to those of course.

File: src/platforms/android/enhance-errors/proguard.mdx [hunk 1]
```diff
@@ -10,6 +10,8 @@ If you want to see de-obfuscated stack traces, you'll need to use [ProGuard](htt
 
 It works the same either way. A (random) UUID will be or has to be generated and placed into the `AndroidManifest.xml`. The same UUID has to be used to upload the mapping file(s). In case a crash happens in the app, the Sentry Android SDK will send the UUID along with the stacktrace to the Sentry server instance. Sentry will then de-obfuscate the stacktrace by searching for the associated mapping file with the same UUID.
 
+If [Source Context](/platforms/android/enhance-errors/source-context) is enable, Sentry can show snippets of code (_source context_) around the location of stack frames, which helps to quickly pinpoint problematic code.
```
Comment (vivianyentran) [inline src/platforms/android/enhance-errors/proguard.mdx:13]:
```suggestion
If [Source Context](/platforms/android/enhance-errors/source-context) is enabled, Sentry can show snippets of code (_source context_) around the location of stack frames, which helps to quickly pinpoint problematic code.
```

File: src/platforms/common/data-management/debug-files/source-context.mdx [hunk 2]
```diff
@@ -65,6 +65,16 @@ For native Android and iOS code, follow the instructions in the [Debug Symbols](
 
 </PlatformSection>
 
+<PlatformSection supported={["android"]}>
+
+<Note>
+
+For Android follow the instructions in the [Android Source Context](/platforms/android/enhance-errors/source-context) guide.
```
Comment (vivianyentran) [inline src/platforms/common/data-management/debug-files/source-context.mdx:72]:
```suggestion
For Android, follow the instructions in the [Android Source Context](/platforms/android/enhance-errors/source-context) guide.
```

**Review by vivianyentran — APPROVED** [review]

**Review by markushi — APPROVED** [review]
LGTM!

**Comment by marandaneto** [comment]
For some reason I can't see the preview, I get `Access Required`.

**Comment by stefanosiano** [comment]
> For some reason I can't see the preview, I get `Access Required`.

Looks like Vercel messed up after last commit. Retriggered and now everything is fine

> @stefanosiano please follow the guideline we agreed on, the source code context should have its own sections in the docs, similar to Flutter/RN, the guidelines have examples of PRs and which files should be modified, not limited to those of course.

I updated some files taking inspiration by the Flutter and iOS PRs, but the source context in its own section looks in contrast with the simplification of the side bar. It's inside the `Enhance Error Reporting` section, but I can move it out if that's the general guideline

**Review by shanamatthews — APPROVED** [review]

**Pull request merged**